### PR TITLE
add `interfaces.generate_legacy_interface` to softmax

### PR DESCRIPTION
**Pull request by imhuay**
The Tensorflow before 1.5 still use the `dim` rather than `axis`, but there are a lot of project still use tf1.4 because of `CUDA8.0`. So I think it should add this change.

If there are any irregularities, please let me know.

### CODE REVIEW

**Comment by imhuay** [comment]
You are right. I just use Keras from 2.1.6.
What I mean is to change 
```
def softmax(x, axis=-1):
    return tf.nn.softmax(x, axis=axis)
```
to 
```
def softmax(x, axis=-1):
    # remove the key parameter
    return tf.nn.softmax(x, axis)
```

How anout this change?
